### PR TITLE
python311Packages.python-idzip init at 0.3.9

### PR DESCRIPTION
**Pull request by ViZiD**
## Description of changes
Seekable, gzip compatible, compression format

Gzip allows to store extra fields in the gzip header. Idzip stores offsets for the efficient seeking there.

https://github.com/bauman/python-idzip

fix bin folder collisions
https://github.com/bauman/python-idzip/pull/23

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by korolr** [comment]
Package is very important to me, you can trust the author

**Comment by mrrva** [comment]
Package is very important to me, you can trust the author

**Comment by thq919** [comment]
Package is very important to me, you can trust the author

File: pkgs/development/python-modules/python-idzip/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  src = fetchFromGitHub {
+    owner = "bauman";
+    repo = "python-idzip";
+    rev = "${version}";
+    hash = "sha256-ChzwC/Afn0qeo5anq4anIu2eI9i6XDnSvB7jAwY7rSw=";
+  };
+
+  patches = [
+    # fix collision
+    ./patches/fix-bin-folder-collisions.patch
```
[the remark this answers is not in the document]
Comment (ViZiD) [reply pkgs/development/python-modules/python-idzip/default.nix]: > Please upstream the patch. This way other distribution could benefit from it.
I used fetchpatch 
is that what I need?

**Comment by ViZiD** [comment]
@natsukium can you review?

File: pkgs/development/python-modules/python-idzip/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    repo = "python-idzip";
+    rev = "refs/tags/${version}";
+    hash = "sha256-ChzwC/Afn0qeo5anq4anIu2eI9i6XDnSvB7jAwY7rSw=";
+  };
+
+  patches = [
+    # fix collision
+    (fetchpatch {
+      name = "fix-bin-folder-collisions.patch";
+      url = "https://github.com/ViZiD/python-idzip/commit/cdb405d3ecca5845a10ce7503f477a4ee6d8db65.patch";
```
Comment (fabaff) [inline pkgs/development/python-modules/python-idzip/default.nix:31]: The patch should come from https://github.com/bauman/python-idzip not a fork. It also helps if the URL to the PR is added to keep track if the patch was merged.

**Review by fabaff — COMMENTED** [review]
Commit message should be `python311Packages.python-idzip init at 0.3.9`.

**Comment by ViZiD** [comment]
@natsukium @fabaff

**Review by natsukium — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by natsukium** [comment]
@ofborg build python311Packages.python-idzip

**Pull request merged**